auth/src/AzureSubscription.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { TokenCredential } from '@azure/core-auth';
+import type { Environment } from '@azure/ms-rest-azure-env';
+import type { AzureAuthentication } from './AzureAuthentication';
+
+/**
+ * A type representing an Azure subscription ID, in the form `${tenantId}/${subscriptionId}`.
+ */
+export type SubscriptionId = string;
+
+/**
+ * A type representing an Azure tenant ID.
+ */
+export type TenantId = string;
+
+/**
+ * Represents an Azure subscription.
+ */
+export interface AzureSubscription {
+    /**
+     * Access to the authentication session associated with this subscription.
+     */
+    readonly authentication: AzureAuthentication;
+
+    /**
+     * The Azure environment to which this subscription belongs.
+     */
+    readonly environment: Environment;
+
+    /**
+     * Whether this subscription belongs to a custom cloud.
+     */
+    readonly isCustomCloud: boolean;
+
+    /**
+     * The display name of this subscription.
+     */
+    readonly name: string;
+
+    /**
+     * The ID of this subscription.
+     */
+    readonly subscriptionId: SubscriptionId;

```

Comment:
This property currently is only the subscription id uuid. Are we intentionally changing it to include the tenant id?

Reply:
I think we should make that change only for the filter setting. I don't know for sure but I think the code here gets just the subscription ID and not the tenant ID.

Reply:
Still, a comment clarifying exactly what is meant would be good.

Reply:
I guess there's conflicting types or comments here. Since `SubscriptionId` type says it includes the tenant id

```ts
/**
 * A type representing an Azure subscription ID, in the form `${tenantId}/${subscriptionId}`.
 */
export type SubscriptionId = string;
```

Reply:
Comment is just plain wrong. I've fixed it.